src/common/types.ts

```diff
@@ -330,6 +330,111 @@ export interface RecordAssemblerOptions<Header extends ReadonlyArray<string>>
   maxFieldCount?: number;
 }
 
+/**
+ * Execution strategy for CSV parsing.
+ *
+ * - `'worker'`: Execute in Worker thread (non-blocking, available on all platforms)
+ * - `'wasm'`: Execute using WebAssembly (high performance, UTF-8 only)
+ *
+ * @remarks
+ * Multiple strategies can be combined by specifying an array.
+ * The order matters: `['worker', 'wasm']` means "execute WASM inside Worker".
+ *
+ * @category Types
+ *
+ * @example Execute in Worker with WASM
+ * ```ts
+ * parse(csv, { execution: ['worker', 'wasm'] })
+ * ```
+ *
+ * @example Execute WASM in main thread
+ * ```ts
+ * parse(csv, { execution: ['wasm'] })
+ * ```
+ */
+export type ExecutionStrategy = "worker" | "wasm";
+
+/**
+ * Execution configuration for CSV parsing.
+ *
+ * @category Types
+ */
+export interface ExecutionOptions {
+  /**
+   * Execution strategy or combination of strategies.
+   *
+   * @default [] (main thread with default implementation)
+   *
+   * @remarks
+   * - Empty array or undefined: Execute in main thread (default)
+   * - Single strategy: `['worker']` or `['wasm']`
+   * - Combined strategies: `['worker', 'wasm']` executes WASM inside Worker
+   *
+   * @example Main thread (default)
+   * ```ts
+   * parse(csv)  // or parse(csv, { execution: [] })
+   * ```
+   *
+   * @example Worker thread
+   * ```ts
+   * parse(csv, { execution: ['worker'] })
+   * ```
+   *
+   * @example WASM in main thread
+   * ```ts
+   * await loadWASM();
+   * parse(csv, { execution: ['wasm'] })
+   * ```
+   *
+   * @example WASM in Worker thread
+   * ```ts
+   * await loadWASM();
+   * parse(csv, { execution: ['worker', 'wasm'] })
+   * ```
+   */
```

Comment:
```suggestion
   * The custom worker must implement the same message protocol as the bundled worker.
   *
   * @example Custom worker
   * ```ts
   * parse(csv, {
   *   engine: {
   *     worker: true,
   *     workerURL: new URL('./custom-worker.js', import.meta.url)
   *   }
   * })
   * ```
```